app/models/grader.rb

```diff
@@ -21,18 +23,29 @@ def self.unique
     select(column_names - ["id"]).distinct
   end
 
-  def grade(assignment, submission)
+  def really_grade(asn_id, sub_id)
+    puts "XX really grade"
+    puts self.inspect
+
+    assignment = Assignment.find(asn_id)
+    submission = Submission.find(sub_id)
+
     ans = do_grading(assignment, submission)
     submission.compute_grade! if submission.grade_complete?
-    ans
+  end
+
+  def grade(assignment, submission, prio = 0)
```

Comment:
The primary use case for this is when I as a professor need to regrade a student's submission immediately, and want to bypass the job queue.